src/litdata/streaming/dataset.py

```diff
@@ -89,7 +92,10 @@ def __init__(
                 If `index_path` is a directory, the function will look for `index.json` within it.
                 If `index_path` is a full file path, it will use that directly.
             force_override_state_dict: Boolean flag for allowing local arguments to override a loaded state dict.
-            transform: Optional transformation function to apply to each item in the dataset.
+            transform: Optional transformation function or list of functions to apply to each item in the dataset.
+            transform_kwargs: Keyword arguments for the transformation function.
+            args: Additional positional arguments.
+            kwargs: Additional keyword arguments.
```

Comment:
It would be nice to also separate this change into a new quick PR, including details, examples, and motivation for the update.